src/storage/file_key_storage.rs

```diff
@@ -0,0 +1,185 @@
+use eframe::Result;
+use enostr::{Keypair, Pubkey, SerializableKeypair};
+
+use crate::Error;
+
+use super::{
+    file_storage::{FileDirectoryInteractor, FileWriterFactory, FileWriterType},
+    key_storage_impl::{KeyStorage, KeyStorageError, KeyStorageResponse},
+};
+
+static SELECTED_PUBKEY_FILE_NAME: &str = "selected_pubkey";
+
+/// An OS agnostic file key storage implementation
+#[derive(Debug, PartialEq)]
+pub struct FileKeyStorage {
+    keys_interactor: FileDirectoryInteractor,
+    selected_key_interactor: FileDirectoryInteractor,
+}
+
+impl FileKeyStorage {
+    pub fn new() -> Result<Self, KeyStorageError> {
+        Ok(Self {
+            keys_interactor: FileWriterFactory::new(FileWriterType::Keys)
+                .build()
+                .map_err(KeyStorageError::OSError)?,
+            selected_key_interactor: FileWriterFactory::new(FileWriterType::SelectedKey)
+                .build()
+                .map_err(KeyStorageError::OSError)?,
+        })
+    }
+
+    fn add_key_internal(&self, key: &Keypair) -> Result<(), KeyStorageError> {
+        self.keys_interactor
+            .write(
+                key.pubkey.hex(),
+                &serde_json::to_string(&SerializableKeypair::from_keypair(key, "", 7))
+                    .map_err(|e| KeyStorageError::Addition(Error::Generic(e.to_string())))?,
+            )
+            .map_err(KeyStorageError::Addition)
+    }
+
+    fn get_keys_internal(&self) -> Result<Vec<Keypair>, KeyStorageError> {
+        let keys = self
```

Comment:
Should just be a more direct style instead of this interactor thing (which is ad-hoc and complects writing and directory operations, so it should be decomposed):

```rust
Directory::new(DataPath::Keys.path()).get_files()
```